crates/cranelift/src/compiler/component.rs

```diff
@@ -280,6 +280,11 @@ impl<'a> TrampolineCompiler<'a> {
                     rets[0] = me.raise_if_negative_one_and_truncate(rets[0]);
                 })
             }
+            Trampoline::ThreadSpawnIndirect { ty: _, table } => {
+                // TODO: eventually pass through the `ty` argument to check the
+                // table's funcref signature.
+                self.translate_thread_spawn_indirect(*table)
```

Comment:
@alexcrichton, I feel like there is some way to actually materialize this `ty` index into a real type so we can check that the `funcref` has the expected type of the thread start function but I don't see it.

Reply:
`TypeFuncIndex` is fine yeah, and the way to pass it to runtime would basically be to embed the constant in the generated code itself. Runtime would then be responsible for translating `TypeFuncIndex` into a `VMSharedTypeIndex` which can be done through various translation tables similar to how core wasm does it

Reply:
Ok, but actually the index being used is coming from a core function type. E.g., in the tests, for `(core type $start (shared (func (param $context i32))))`, we use `$start` as the index passed through to this canonical built-in. I see how to retrieve a `TypeFunc` using `ComponentTypes` but I don't think that's what we actually want here &mdash; that would be a component type. Is `TypeFuncIndex` the right index to be using here? How do we get to its core type (presumably backed by `VMSharedTypeIndex`)?

Reply:
Agreed yeah we don't want to deal with `TypeFunc` here as the type-check should be a single comparison ideally, via `VMShareTypeIndex`. Converting from `TypeFuncIndex` will require a table of some form stored in `VMComponentContext` which is filled in on initialization. That happens for core modules, for example, where basically `VMContext` has a table from `TypeFuncIndex` to `VMSharedTypeIndex`